Add support of Mistral-7B into TT-Transformers

pprajapatiTT opened the pull request:
### Ticket
https://github.com/tenstorrent/tt-metal/issues/19490

### Problem description
Add support of Mistral-7B into TT-Transformers and deprecates the old stale (lower-perf) implementation.

### What's changed
- Remove deprecated implementation of Mistral-7B (in `/models/demos/wormhole/mistral-7b/)
- Add Mistral-7B support to TT-Transformers
  - Validated Mistral-7B-Instruct-v0.3 using TT-Transformers via HuggingFace interface
  - Add performance and accuracy numbers
  - Add CI tests

### Checklist
- [ ] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes
- [ ] [All single-device tests](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select.yaml)
- [ ] [All T3K tests](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-t3k.yaml)

pprajapatiTT committed:
Add support of Mistral-7B into TT-Transformers

Also deprecate the old stale implementation.

Signed-off-by: Pratikkumar Prajapati <pprajapati@tenstorrent.com>

pprajapatiTT committed:
Add support of Mistral-7B into TT-Transformers

Also deprecate the old stale implementation.

Signed-off-by: Pratikkumar Prajapati <pprajapati@tenstorrent.com>

mtairum committed:
Add Mistral-7B to CI dispatch tests

mtairum committed:
Add Mistral-7B to CI dispatch tests

tt-rkim commented on tests/scripts/run_performance.sh:
```diff
@@ -60,7 +60,6 @@ run_perf_models_llm_javelin() {
     fi
 
     env pytest -n auto models/demos/falcon7b_common/tests -m $test_marker
-    env pytest -n auto models/demos/wormhole/mistral7b/tests -m $test_marker
```
Whoa whoa whoa - we're taking mistral7b out of all single-card pipelines, but still list it as a README model for single-card? What's up with that?

mtairum replied:
@tt-rkim This PR was supposed to be in draft mode :) 

I'm adding the new mistral-7B onto tests again and when that's in and CI pipelines are passing, it will be marked for review.

mtairum committed:
Update Mistral-7B single-device CI tests and delete obselete tests

mtairum committed:
Update Mistral-7B single-device CI tests and delete obselete tests

mtairum committed:
Add Mistral-7B to all-post-commit CI

mtairum committed:
Add Mistral-7B to all-post-commit CI

mtairum committed:
Fix how we get the model name for HF models in TT-Transformers

mtairum committed:
Fix how we get the model name for HF models in TT-Transformers

pprajapatiTT committed:
Add T3K tests for Mistral 7B

Signed-off-by: Pratikkumar Prajapati <pprajapati@tenstorrent.com>

pprajapatiTT committed:
Add T3K tests for Mistral 7B

Signed-off-by: Pratikkumar Prajapati <pprajapati@tenstorrent.com>

mtairum commented on tests/scripts/t3000/run_t3000_demo_tests.sh:
```diff
@@ -125,6 +125,28 @@ run_t3000_falcon7b_tests(){
+
+  echo "LOG_METAL: Running run_t3000_mistral7b_demo_tests"
+
+  export WH_ARCH_YAML=wormhole_b0_80_arch_eth_dispatch.yaml
```
Please don't export these variables, otherwise this will pollute other tests.

Just save them as variables and append them to the run command, like we do in the other models.

pprajapatiTT committed:
Merge branch 'main' into pprajapati-19490

mtairum committed:
Added Mistral T3K tests to the yaml scripts

mtairum committed:
Added Mistral T3K tests to the yaml scripts

mtairum committed:
Rename mistral7B tests

mtairum committed:
Rename mistral7B tests

pprajapatiTT committed:
rename remaining test from mistral7b to mistral

Signed-off-by: Pratikkumar Prajapati <pprajapati@tenstorrent.com>

pprajapatiTT committed:
rename remaining test from mistral7b to mistral

Signed-off-by: Pratikkumar Prajapati <pprajapati@tenstorrent.com>

pprajapatiTT committed:
remove 7b from mistral test name in perplexity-tests yaml

Signed-off-by: Pratikkumar Prajapati <pprajapati@tenstorrent.com>

pprajapatiTT committed:
remove 7b from mistral test name in perplexity-tests yaml

Signed-off-by: Pratikkumar Prajapati <pprajapati@tenstorrent.com>

pprajapatiTT committed:
Add protobuf==3.20.0 in requirements dep

Signed-off-by: Pratikkumar Prajapati <pprajapati@tenstorrent.com>

pprajapatiTT committed:
Add protobuf==3.20.0 in requirements dep

Signed-off-by: Pratikkumar Prajapati <pprajapati@tenstorrent.com>

pprajapatiTT committed:
Add model wise PCC expectations

Signed-off-by: Pratikkumar Prajapati <pprajapati@tenstorrent.com>

pprajapatiTT committed:
Add model wise PCC expectations

Signed-off-by: Pratikkumar Prajapati <pprajapati@tenstorrent.com>

pprajapatiTT committed:
fix normalized_path for HF model

Signed-off-by: Pratikkumar Prajapati <pprajapati@tenstorrent.com>

pprajapatiTT committed:
fix normalized_path for HF model

Signed-off-by: Pratikkumar Prajapati <pprajapati@tenstorrent.com>

pprajapatiTT committed:
Add protobuf==3.20.0 in requirements-dev.txt

Signed-off-by: Pratikkumar Prajapati <pprajapati@tenstorrent.com>

pprajapatiTT committed:
Add protobuf==3.20.0 in requirements-dev.txt

Signed-off-by: Pratikkumar Prajapati <pprajapati@tenstorrent.com>

sraizada-tt committed:
changes to llama3_subdevices for vllm integration  (#19845)

### What's changed
updates to generator and mode switching for vllm integration 

### Checklist
- [ ] [All post
commit][(https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI
passes](https://github.com/tenstorrent/tt-metal/actions/runs/14132258580)
TG frequent:
https://github.com/tenstorrent/tt-metal/actions/runs/14132276364
TG demo:
https://github.com/tenstorrent/tt-metal/actions/runs/14132287889

tt-aho committed:
#0: Fix lowest_occupied_compute_l1_address when sub-devices are enabled (#19240)

Previously we only checked the L1 allocations of specific sub-devices
allocators that were specified, but we should also query the global
allocator as they are part of the total l1 address space

### Ticket
Link to Github Issue

### Problem description
CB validation was not working as expected, as we were only checking the
sub-device allocators, and not the global allocator as well.

### What's changed
Check the global allocator in addition to any specified local
allocators. Add unit tests to validate the CB validation is verifying
against both.

### Checklist
- [x] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
https://github.com/tenstorrent/tt-metal/actions/runs/13913911095
- [ ] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI passes (if applicable)
- [ ] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [ ] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
CI passes (if applicable)
- [x] New/Existing tests provide coverage for changes

ayerofieiev-tt committed:
[skip ci] Update CODEOWNERS (#19850)

### Ticket
None

### Problem description
Add @ayerofieiev-tt to codeowners in tt-train

### What's changed
Added

rfurko-tt committed:
[BugFix][tt-train] Add missing value to config that was introduced in previous PR  (#19847)

### Problem description
LLaMa integration added dependency on `model_type` config key.
Unfortunately, none of the gpt2 configs have this parameter set.

### What's changed
Add `model_type` to configs.

### Checklist
- [x] New/Existing tests provide coverage for changes

mbahnasTT committed:
[skip ci] Update Yolov4 web demo README.md (#19808)

mtairum committed:
Fix LLama-70B TG ND hang on lm-head and add support for Deepseek R1 LLama-70B distilled weights (#19853)

### Ticket
https://github.com/tenstorrent/tt-metal/issues/19723


### What's changed
- Fix TG ND hang on lm-head by adding a specific program config (had to
reduce grid size to (7, 8).
- Add support for HF weights to the TG codebase, specifically: Deepseek
R1 LLama-70B distilled weights

jbaumanTT committed:
#0: Improve test_pgm_dispatch (#18784)

### Problem description
Sometimes tests may run with the AI clock throttled, and it can be hard
to notice that after-the-fact.

### What's changed
In particular add new tests that seemed helpful when doing a cost model
calculation (such as having CBs covering portions of the workers, and
CRTAS),
as well as a flag to add a lot of random tests. Also add a flag to
report
benchmark parameters in the JSON to make it possible for later scripts
to do
further analysis.

Add the current clock frequency to the output so we can detect
throttling more
easily.

Also do CloseDevice on some early returns, so later tests that attempt
to use
the same device don't run into problems.

### Checklist
- [x] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
- [ ] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI passes (if applicable)
- [ ] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [ ] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
CI passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

williamlyTT committed:
#0: [skip ci] Add missing inputs for "(Single-card) Tests for new models" workflow (#19846)

### Ticket
Link to Github Issue

### Problem description
Workflow is missing the build docker/wheel/build-artifact inputs to
`single-card-demo-tests-impl.yaml`

### What's changed
Added the missing inputs in.

### Checklist
- [ ] **(For models and ops writers)** Full [new models
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
CI passes (if applicable)
https://github.com/tenstorrent/tt-metal/actions/runs/14133203027

williamlyTT committed:
#0: [skip ci] Fix L2+ttnn sweep artifact and wheel inputs to setup-job (#19854)

### Ticket
Link to Github Issue

### Problem description
Metal L2 nightly is not a reusable/impl workflow so it doesn't have
build and wheel artifact names as inputs. They're passed in directly
from the build-artifact job.
Also fixed ttnn - run sweeps which has the same issue.

### What's changed
Change inputs to setup job step to the expected parameters.

### Checklist
- [ ] Metal L2:
https://github.com/tenstorrent/tt-metal/actions/runs/14133411682
- [ ] ttnn sweeps (`add` module only):
https://github.com/tenstorrent/tt-metal/actions/runs/14133562179

keerthana-r-mcw committed:
Yolo evaluation script update (#19617)

### Ticket
https://github.com/tenstorrent/tt-metal/issues/19613
### Problem description
Modify the models/experimental/yolo_evaluation script to support both
320 and 640 resolution sizes and use the trace script for inference in
yolov4.

### Checklist
- [x] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/runs/14060029639)
CI passes

---------

Co-authored-by: Mohamed Bahnas <mbahnas@tenstorrent.com>

skrsticTT committed:
Skip hanging conv2d tests (#19852)

### Ticket
#19831

### Problem description
2 conv2d tests are hanging on blackhole - adding temporary skip until
issue above is not closed.

blozano-tt committed:
[skip ci] Update build-wrapper.yaml to build 22.04 and 24.04 configs (#19860)

For migration from 20.04 -> 22.04 and beyond!

https://github.com/tenstorrent/tt-metal/actions/runs/14134538925

blozano-tt committed:
[skip ci] Make build-artifact workflow default to ubuntu 22.04 (#19864)

See title

blozano-tt committed:
Don't default to libc++, use libstdc++ now (#19807)

### Ticket
Closes #19524 

@marty1885 - this one is for you! (Hopefully it lands)

### Problem description
libc++ is the non default c++ standard library implementation.
It makes it hard to mix metal binaries with other system libraries if we
don't use the `standard` c++ library.

### What's changed
Switch to what is standard.

### Checklist
- [x] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/runs/14133945157)
CI passes
- [x] [ttnn unit tests after skip ttnn test and
rebase](https://github.com/tenstorrent/tt-metal/actions/runs/14135994938)
- [x] New/Existing tests provide coverage for changes

mbahnasTT committed:
[skip ci] Update README.md ViT perf (#19871)

Update ViT perf to be 1100 FPS instead of 912 FPS.

mbahnasTT committed:
Mbahnas/vit mv to demos (#19665)

### Ticket
Link to Github Issue

### Problem description
Moving ViT model from experimental to demos, based on the discussion in
https://github.com/tenstorrent/tt-metal/pull/19511#discussion_r2019214444

### What's changed
Moving ViT model from `models/experimental/functional_vit` to
`models/demos/vit`, and deleting the obsolete `models/experimental/vit`

### Checklist
- [ ] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
- [ ] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI passes (if applicable)
- [ ] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [ ] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
CI passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

ubcheema committed:
Add support for line mcast (#19823)

blozano-tt committed:
Add missing includes (#19876)

SeanNijjar committed:
fix build issue (#19874)

Not sure how this passed before... Fix kernel build issue for ring
fabric



### Checklist
- [x] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI: https://github.com/tenstorrent/tt-metal/actions/runs/14139382236
- [x] metal microbench:
https://github.com/tenstorrent/tt-metal/actions/runs/14139367265
- [ ] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI passes (if applicable)
- [ ] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [ ] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
CI passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

tt-aho committed:
#0: Fix in calculating ptr in eth_send_packet_byte_addr (#19875)

### Ticket
Link to Github Issue

### Problem description
ptr offset calculation does not take into account q_num.

### What's changed
Add offset based on q_num to ptr addr.

### Checklist
- [x] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
https://github.com/tenstorrent/tt-metal/actions/runs/14139035129
- [ ] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI passes (if applicable)
- [ ] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [ ] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
CI passes (if applicable)
- [x] New/Existing tests provide coverage for changes

sminakov-tt committed:
Fix race condition in Moreh use-case tests and enable them on CI (#19880)

### Ticket

### Problem description
There is a race condition in Moreh tests
(test_ccl_multi_cq_multi_device) which is caused by filling the same
host buffer with different values on multiple threads.

### What's changed
Fixed the race condition and enable the test on T3K CI

### Checklist
- [x] [All post commit CI
passes](https://github.com/tenstorrent/tt-metal/actions/runs/14143611757)
- [x] [T3K unit tests CI
passes](https://github.com/tenstorrent/tt-metal/actions/runs/14143614781)
- [x] New/Existing tests provide coverage for changes

KalaivaniMCW committed:
#15132: Improve ttnn.tanh accuracy  (#19334)

### Ticket
Link to Github Issue #15132

### Problem description
ttnn.tanh has low PCC for input range  (-3.5, 3.5)

### What's changed
- An alternate implementation that is not dependant on the look-up table
LLK implementation to provide better PCC necessary for models.
- overall the accuracy for range (-3, 3) has improved with the PCC
greater than 0.9999 for BFLOAT16 dtype

As per @amalbasaTT 's comment the new improved PCC on using this
implementation
albert_v2_base: 0.9666459989505328
albert_v2_large: 0.9673880460396883
albert_v2_xlarge: 0.9682215217344369
albert_v2_xxlarge: 0.9918737729342227

### Follow-up: 
- Convert this composite implementation into a device operation
https://github.com/tenstorrent/tt-metal/issues/19461

### Checklist
- [x] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
https://github.com/tenstorrent/tt-metal/actions/runs/13971672604
https://github.com/tenstorrent/tt-metal/actions/runs/14139810091
- [ ] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI passes (if applicable)
- [x] Sweeps -
https://github.com/tenstorrent/tt-metal/actions/runs/13991964779
- [ ] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [ ] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
CI passes (if applicable)
- [x] New/Existing tests provide coverage for changes

afuller-TT committed:
Afuller/fix clang tidy (#19885)

Clang Tidy is broken.  Fix it.

aagarwalTT committed:
[Fabric] Minor test fixes (#19867)

### Ticket
Link to Github Issue

### Problem description
Provide context for the problem.

### What's changed
1. Fix client interface in mcast test
2. Remove header from 1d tx kernel

### Checklist
- [x] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
(https://github.com/tenstorrent/tt-metal/actions/runs/14137198182)
- [ ] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI passes (if applicable)
- [ ] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [ ] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
CI passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

afuller-TT committed:
Fix clang tidy issues (#19886)

ClangTidy is broken.  Fix it.

skrsticTT committed:
Remove blackhole condition for asserts on l1 usage (#19866)

### Ticket
#17216

### What's changed
Remove blackhole condition for asserts on l1 usage, as asserts are valid
for blackhole too.
Made sure that conv2d UTs and sweeps passed with a change.

### Checklist
- [x] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/runs/14136120226)

skrsticTT committed:
Move conv2d tests from all post-commit pipeline to nightly pipeline (#19702)

### Ticket
#19701

### What's changed
As number of conv2d unit tests grows, move majority of these to nightly
pipeline to relax all post-commit pipeline.

### Checklist
- [x] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/runs/14085062047)
- [x] [Nightly tt-metal L2
tests](https://github.com/tenstorrent/tt-metal/actions/runs/14085074054)

VirdhatchaniKN committed:
#6583: Fix acosh_bw (#19695)

### Ticket
https://github.com/tenstorrent/tt-metal/issues/6583

### Problem description
Acosh backward fails for edge cases when input, gradient is in range
[0,1]

### What's changed

- Update `acosh_bw` Logic to match PyTorch API
- As mentioned
[here](https://github.com/tenstorrent/tt-metal/pull/11243#discussion_r1808816083),
used nan_to_num to compare Nan to numbers.
  - Test passes now in both BH , WH

Updated acosh backward logic vs current logic is as below :

<img width="635" alt="Screenshot 2025-03-27 at 6 32 02 PM"
src="https://github.com/user-attachments/assets/12a2f660-5aa5-44d1-bf82-8f64038c9262"
/>


### Checklist
- [x] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/runs/14134224476)
CI passes
- [x] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/runs/14109838293)
CI passes as in main
- [x] https://github.com/tenstorrent/tt-metal/actions/runs/14164235499
CI passes for acosh_bw sweep test
- [x] https://github.com/tenstorrent/tt-metal/actions/runs/14164230219
CI passes for acosh_bw sharded sweep test

pavlejosipovic committed:
Conv2d weights kernel cleanup (#19900)

Cleanup conv2d weights kernels

* Remove code which supports writing output to interleaved tensor (as
it's not used).
* Remove four kernel reading for reading weights, rely on a bit
generalised kernel used for previously in case split reader was enabled.
* Reduce number of compile and runtime args passed in to weights reader
kernels.

### Checklist
- [x] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/runs/14160149371)
CI passes
- [x] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/runs/14160154467)
CI passes
- [x] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/runs/14160151689)
CI passes
- [x] [(Single-card) Frequent model and ttnn
tests](https://github.com/tenstorrent/tt-metal/actions/runs/14160173461)
CI passes
- [x] Nightly tt-metal L2 tests
(https://github.com/tenstorrent/tt-metal/actions/runs/14160173461)

Co-authored-by: Pavle Josipovic <pjosipovic@tenstorrent.com>

fvranicTT committed:
chore: intake llk changes - March 28, 2025 (#19855)

### Problem description
This PR serves as integration of llk changes until Friday 28th 2025.

### What's changed
The following changes are included:
- https://github.com/tenstorrent/tt-llk/pull/54 (test: reset card after
each test run)
- https://github.com/tenstorrent/tt-llk/pull/59 ([Test Infra] Added
Support for Independent Source Register Formats and Custom Test
Combinations)
- https://github.com/tenstorrent/tt-llk/pull/63 (#16439: Disable
instruction coalescing feature)
- https://github.com/tenstorrent/tt-llk/pull/64 (build: remove unused
python imports using autoflake)
- https://github.com/tenstorrent/tt-llk/pull/66 ([Test Infrastructure]
Upgraded Test Interface and Output of Results)
- https://github.com/tenstorrent/tt-llk/pull/65 (ci: enable ci build)
- https://github.com/tenstorrent/tt-llk/pull/70 (Cleanup of superfluous
includes and removal of `using namespace sfpi`)
- https://github.com/tenstorrent/tt-llk/pull/68 ([Test Infrastructure]
Fixed bug in test_tilize_calculate_untilize_L1)

### Checklist
- [x] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/runs/14134174798)
CI passes
- [x] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/runs/14134188837)
CI passes (if applicable)

nathan-TT committed:
#0: sfpi update v6.7.0 (#19909)

Update https://github.com/tenstorrent/sfpi/releases/tag/v6.7.0

mtairum committed:
[Llama3-8B Blackhole] Change SDPA decode chunk size to 128 to avoid hang in BH and add demo to CI (#19794)

### Ticket
https://github.com/tenstorrent/tt-metal/issues/18135

### What's changed
Updated Llama3 config to avoid demo hangs on 1xP150

### Checklist
- [x] [BH Demo
pipeline](https://github.com/tenstorrent/tt-metal/actions/runs/14113251816)
- [x] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/runs/14113269369)
- [x] [BH Post
commit](https://github.com/tenstorrent/tt-metal/actions/runs/14113281560)

johanna-rock-tt committed:
Add measurement option to decode demo (#19863)

### Problem description
No option to run device perf measurements without changes in code.

### What's changed
Added and merged options so that we have options to run: quick, full,
stress-test and measure-device-perf runs.

- [x] TG demo tests:
https://github.com/tenstorrent/tt-metal/actions/runs/14135281013
- [x] TG stress tests:
https://github.com/tenstorrent/tt-metal/actions/runs/14156099074

### Checklist
- [ ] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
- [ ] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI passes (if applicable)
- [ ] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [ ] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
CI passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

pavlejosipovic committed:
Fix tt-forge conv1d cases (#19905)

* Allow conv1d to use width and block sharding
* Add auto-shard support for these cases
* Skip conv1d deptwise code path in case conv has bias, as conv1d
deptwise doesn't have bias
support ATM.

### Checklist
- [x] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/runs/14169195836)
CI passes
- [x] [(Single-card) Frequent model and ttnn
tests](https://github.com/tenstorrent/tt-metal/actions/runs/14169208808)
CI passes
- [x] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/runs/14169202954)
CI passes
- [x] [(Single-card) Device perf
regressions](https://github.com/tenstorrent/tt-metal/actions/runs/14171444944)

Co-authored-by: Pavle Josipovic <pjosipovic@tenstorrent.com>

jenperson committed:
[skip ci] Documentation tweaks (#19469)

### Ticket
Link to Github Issue
N/A docs updates

### Problem description
Provide context for the problem.

I simplified some of the language of the Getting Started guides and
fixed grammatical errors. I also ensured titles all used the same case
style, which is title case.

### What's changed
Describe the approach used to solve the problem.
Summarize the changes made and its impact.

### Checklist
- [ ] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
- [ ] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI passes (if applicable)
- [ ] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [ ] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
CI passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

mradosavljevicTT committed:
Resnet50 multichip demos (#19260)

### Ticket
  #17919 

### Problem description
Created demos for resnet on n300, t3k, galaxy. Added tests to CI.

### Checklist
- [x] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/runs/14134404487)
CI passes
- [x] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/runs/14134410365)
CI passes
- [x] [(Single card) Model performance
tests](https://github.com/tenstorrent/tt-metal/actions/runs/14134416903)
CI passes
- [x] [(Single card) Demo
tests](https://github.com/tenstorrent/tt-metal/actions/runs/14134455713)
Same test is failing on
[main](https://github.com/tenstorrent/tt-metal/actions/runs/14122962891)
- [x] [(T3K) Demo
tests](https://github.com/tenstorrent/tt-metal/actions/runs/14134420117)
Same test is failing on
[main](https://github.com/tenstorrent/tt-metal/actions/runs/14119220112)
- [x] [(T3K) Unit
tests](https://github.com/tenstorrent/tt-metal/actions/runs/14134422366)
Same test is failing on
[main](https://github.com/tenstorrent/tt-metal/actions/runs/14131843853)

djordje-tt committed:
#0: Increase ring buffer to 120KB to reduce dispatch time (#19840)

### Problem description
SDPA Kernel size is ~50KB which slows down dispatch due to not enough
space in ring buffer. By increasing ring buffer from 69KB -> 120KB
dispatch time drops to 0.8us which is current minimum for dispatch time
between 2 ops.

### What's changed
Currently we are using environment variable flag
TT_METAL_WORKER_RINGBUFFER_SIZE to modify ring buffer but once runtime
team produces proper implementation for ringbuffer size parametrization
we will make transition. Up to then we need to use env_variable.

### Checklist
- [ ] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/runs/14130528761)
CI passes
- [ ] [TG
Demo](https://github.com/tenstorrent/tt-metal/actions/runs/14130366411)
- [ ] [TG
Frequent](https://github.com/tenstorrent/tt-metal/actions/runs/14130589644)

philei-tt committed:
Move copy_values and load_immediate_values to llk API (#19463)

### Ticket
Github Issue: https://github.com/tenstorrent/tt-metal/issues/19350 

### Problem description
`copy_values` and `load_immediate_values` are llk functions added in
this PR https://github.com/tenstorrent/tt-metal/pull/19017, which might
be useful for other developers. That's why I want to make them part of
LLK API.

### What's changed
New
[file](https://github.com/tenstorrent/tt-metal/blob/philei/sfpu_common/tt_metal/include/compute_kernel_api/common_sfpu.h)
under `compute_kernel_api` with four functions (`load_immediate_values`,
`copy_values`, and their inits):

```c++
/**
 * Loads an immediate floating point value into the specified DST register.
 *
 * The DST register buffer must be in acquired state via *tile_regs_acquire* call. This call is blocking and is only
 * available on the compute engine.
 *
 * Return value: None
 *
 * | Argument       | Description                                                           | Type     | Valid Range |
 * Required |
 * |----------------|-----------------------------------------------------------------------|----------|-------------------------------------------------------|----------|
 * | idst           | The index of the tile in DST register buffer to load the value into   | uint32_t | Must be less
 * than the size of the DST register buffer | True     | | val            | The floating point value to load | float |
 * Any valid floating point value                        | True     |
 */
ALWI void load_immediate_value(uint32_t idst, float val) {
    MATH(sfpu::llk_math_eltwise_unary_sfpu_load_imm(idst, val));
}

/**
 * Copies all values from the tile in idst1 to the tile in idst0 in the DST register buffer.
 *
 * The DST register buffer must be in acquired state via *tile_regs_acquire* call. This call is blocking and is only
 * available on the compute engine.
 *
 * Return value: None
 *
 * | Argument       | Description                                                           | Type     | Valid Range |
 * Required |
 * |----------------|-----------------------------------------------------------------------|----------|-------------------------------------------------------|----------|
 * | idst0          | The index of the tile in DST register buffer to copy values to        | uint32_t | Must be less
 * than the size of the DST register buffer | True     | | idst1          | The index of the tile in DST register buffer
 * to copy values from      | uint32_t | Must be less than the size of the DST register buffer | True     |
 */
ALWI void copy_values(uint idst0, uint32_t idst1) {
    MATH(sfpu::llk_math_eltwise_binary_sfpu_copy_values(idst0, idst1));
}

```

### Checklist
- [x] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
(https://github.com/tenstorrent/tt-metal/actions/runs/14173607149)
- [ ] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI passes (if applicable)
(https://github.com/tenstorrent/tt-metal/actions/runs/14147814506)
- [ ] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [ ] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
CI passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

---------

Co-authored-by: Pavlo Hilei <pavlo.hilei@gmail.com>

blozano-tt committed:
Add `include-what-you-use` to docker image (#19903)

include-what-you-use seems to have potential for helping us develop
better software.

for now, it can be a development aid.

it could also become used in a workflow.

wenbinlyuTT committed:
Add support for scale-as-tensor in quant ops (#19597)

### Ticket
#15934

### Problem description
Allowing the user to pass the quantization scale as a tensor.
This is a small functional addition, but it involves a few bug fixes in
BinaryNg and warrants extensive unit testing.
This also paves the way to supporting per-channel quantization, in which
both scale and zero-point must be passed as tensors.

### What's changed
Allowing the user to pass the quantization scale as a scalar-tensor by
modifying the existing API entries to use `std::variant`.
Fix the incorrect assumption in BinaryNg that tensor A and scalar B
always have the same data type/format, as well as the messy logic around
data types/formats.
When the scale is passed as a scalar, force it to be f32 to improve
accuracy and bypass #19453
When the scale is passed as a tensor, switch to a composite op fallback
(accuracy and performance are not as good, and must workaround issues
like #18955)
Add more quantization-related unit tests and parameter combinations,
also add new assertions that catch cases missed by PCC (output tensor is
uniform).

### Checklist
- [ ] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
- [ ] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI passes (if applicable)
- [ ] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [ ] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
CI passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

---------

Signed-off-by: wenbinlyuTT <wenbinlyu@tenstorrent.com>

aliuTT committed:
ControlPlane: cleanup support for Torus on UBB (#19827)

### Ticket
[Link to Github
Issue](https://github.com/tenstorrent/tt-metal/issues/19826)

### Problem description
Various clean up needed to support launching control plane on UBB with
Torus connections.

### What's changed
Read asic ids for ubbs to determine corner chips.

### Checklist
- [ ] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
- [ ] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI passes (if applicable)
- [ ] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [ ] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
CI passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

philei-tt committed:
Make ttnn.squeeze great again: unify behavior with torch (#19747)

### Ticket
Github Issue https://github.com/tenstorrent/tt-metal/issues/19746 

### Problem description
`ttnn.squeeze` doesn't support `dim=None` and `dim=[...]` cases, like
[torch.squeeze](https://pytorch.org/docs/stable/generated/torch.squeeze.html)
does.

### What's changed
- If `ttnn.squeeze` is not provided with dim (or dim=None explicitly),
is squeezes all 1-sized dimensions
- if `ttnn.squeeze` is provided with list of dims, then it tries to
squeeze those dimensions
- Added more checks to have similar errors to pytorch
- Added more tests (dim=None, dim=[...], errors) and extend shapes for
available tests (0 rank handling)
- As a small bonus I fixed printing 0-ranked tensors

### Checklist
- [x] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
(https://github.com/tenstorrent/tt-metal/actions/runs/14145537850)
- [x] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI passes (if applicable)
(https://github.com/tenstorrent/tt-metal/actions/runs/14145537850)
- [ ] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [ ] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
CI passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

---------

Co-authored-by: Pavlo Hilei <pavlo.hilei@gmail.com>

skhorasganiTT committed:
[skip ci] Update codeowners with additional entries for unowned models folders (#19511)

### Ticket
N/A

### Problem description
- Many models folders do not have assigned codeowners

### What's changed
- Created new entries for unassigned models folders and updated
codeowners for a few existing folders

### Checklist
- [x] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
- [x] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI passes (if applicable)
- [x] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [x] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
- [x] **(For models and ops writers)** Full [new models
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
CI passes (if applicable)
- [x] New/Existing tests provide coverage for changes

cc @uaydonat

---------

Signed-off-by: Salar Hosseini <skhorasgani@tenstorrent.com>

sagarwalTT committed:
Improve Compile Time Args to Make Them Programmatically Accessible (#18379)

### Ticket
#16040

### Description
This PR enables compile time args to be programmatically accessible,
which allows for improved code reuse when accessing them.

Consider the following code:
```
#ifndef SINGLE_TENSOR
#if defined(TENSOR1_SHARDED_MEM_LAYOUT)
#if defined(TENSOR0_SHARDED_MEM_LAYOUT)
constexpr sharded_addrgen_fields in1_sharded_addrgen_fields = {
    true,
    get_compile_time_arg_val(17),
    get_compile_time_arg_val(18),
    get_compile_time_arg_val(19),
    get_compile_time_arg_val(20),
    get_compile_time_arg_val(21),
    get_compile_time_arg_val(22),
    get_compile_time_arg_val(23) != 0};
#else
// Then we are only consuming ct args for second operand and we resume from operand 10
constexpr sharded_addrgen_fields in1_sharded_addrgen_fields = {
    true,
    get_compile_time_arg_val(10),
    get_compile_time_arg_val(11),
    get_compile_time_arg_val(12),
    get_compile_time_arg_val(13),
    get_compile_time_arg_val(14),
    get_compile_time_arg_val(15),
    get_compile_time_arg_val(16) != 0};
#endif
#else
constexpr sharded_addrgen_fields in1_sharded_addrgen_fields = {0, 0, 0, 0, 0, 0, 0, 0};
#endif
#endif
```

With this PR, the above can be simplified:
```
// some CT arg capture above
constexpr operand_1_ct_arg_idx_start = last_ct_arg + 1; // (last_ct arg also constexpr)
#ifndef SINGLE_TENSOR
constexpr sharded_addrgen_fields in1_sharded_addrgen_fields = {
    true,
    get_compile_time_arg_val(operand_1_ct_arg_idx_start),
    get_compile_time_arg_val(operand_1_ct_arg_idx_start + 1),
    get_compile_time_arg_val(operand_1_ct_arg_idx_start + 2),
    get_compile_time_arg_val(operand_1_ct_arg_idx_start + 3),
    get_compile_time_arg_val(operand_1_ct_arg_idx_start + 4),
    get_compile_time_arg_val(operand_1_ct_arg_idx_start + 5),
    get_compile_time_arg_val(operand_1_ct_arg_idx_start + 6) != 0};
#else
constexpr sharded_addrgen_fields in1_sharded_addrgen_fields = {0, 0, 0, 0, 0, 0, 0, 0};
#endif
```

And now this simplified version can be converted to its own function and
be reused across kernels:
```
template <size_t start_idx>
constexpr sharded_addrgen_fields build_sharded_addrgen_fields_from_ct_args() {
  return sharded_addrgen_fields{
    true,
    get_compile_time_arg_val(start_idx),
    get_compile_time_arg_val(start_idx + 1),
    get_compile_time_arg_val(start_idx + 2),
    get_compile_time_arg_val(start_idx + 3),
    get_compile_time_arg_val(start_idx + 4),
    get_compile_time_arg_val(start_idx + 5),
    get_compile_time_arg_val(start_idx + 6) != 0};
}
```

On the host side, instead of passing in kernel compile time args with
the format `KERNEL_COMPILE_TIME_ARG_<idx>=<val>`, they are now passed in
as `KERNEL_COMPILE_TIME_ARGS=<ct_arg_val_0>,<ct_arg_val_1>,...`. On the
device side, the values in `KERNEL_COMPILE_TIME_ARGS` are stored in
`kernel_compile_time_args` which is a `std::array`. The macro
`get_compile_time_arg_val(idx)` returns the value stored in the
corresponding index of this array. The array can also be directly
accessed in kernels if needed.

If the kernel compile time args are only being accessed at compile time,
the args don't take up any memory and so the only limit to the number of
compile time args is the maximum command length that the terminal can
handle. However, if any compile time args are being accessed at runtime,
the ENTIRE contents of `kernel_compile_time_args` will be stored in
memory which will significantly limit the number of compile time args a
kernel can take in.

Spot-checking a few kernel binaries with/without this change shows no
difference in their size.

As part of this PR, I've added a test,
`DeviceFixture.TensixTestTwentyThousandCompileTimeArgs`, that creates a
kernel with 20,000 compile time args passed in. In this test, the kernel
sums up the values of all the compile time args at compile time. The
only reason I couldn't pass more compile time args is because I was
hitting the maximum command length. In a previous version of the test, I
was accessing the compile time args at runtime in the kernel and I was
only able to pass in a couple hundred compile time args before hitting a
memory limit.

### Checklist
- [ ] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
- [ ] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI passes (if applicable)
- [ ] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [ ] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
CI passes (if applicable)
- [x] New/Existing tests provide coverage for changes

yugaoTT committed:
Use NoC0 in fabric worker  (#19861)

The noc congestion is resolved by changing to noc0 in worker. 

### Checklist
- [x] [All post commit]
https://github.com/tenstorrent/tt-metal/actions/runs/14134754376
- [x] ubench
https://github.com/tenstorrent/tt-metal/actions/runs/14173342880/job/39702585300
- [x] t3k unit
https://github.com/tenstorrent/tt-metal/actions/runs/14173351555/job/39702629459#logs

jbaumanTT committed:
#19836: Add support for reusing programs across different sub-device … (#19930)

…managers

### Ticket
#19836 

### Problem description
Currently, running the same program under different subdevice managers
is disallowed.

### What's changed
We need to cache different program command sequence for each sub-device
manager, since the command sequence bakes in information about the
subdevice the program is running on.

Using the same program on multiple sub-device managers makes dispatch
slightly less efficient, since RTAs may need to be copied from one
command sequence into another whenever the program is enqueued.

### Checklist
- [x] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
- [ ] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI passes (if applicable)
- [ ] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [ ] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
CI passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

esmalTT committed:
Add support for in-place halo (#19834)

### Ticket
#14450

### Problem description
Currently separate buffers are used for the input and output shards of
the Halo Op. This should be adjusted so the output shard contains the
input shard, thereby making Halo an "in-place" operation.

This was originally merged/reviewed in
https://github.com/tenstorrent/tt-metal/pull/18329, but the commit was
reverted due to regressions on BH and WH tests. These issues have been
fixed by correcting the assumptions about the core grid shape and with
with the worker NOC coordinates used to specify the destination of the
`noc_semaphore_inc` calls.

### What's changed
- In place flags were added to Conv2D and Maxpool, setting these to true
prompts deallocation of the sharded input tensor to prompt re-allocation
with output size
- Scratch tensor/CB were created for temporary storage of remote sticks
- New in place version of halo gather kernel was implemented
- New kernel functions were implemented to copy remote sticks to/from
scratch buffer
- Gather kernel was updated to perform remote / local data movement on
same core to block local copies until after scratch writes with in place
operation
- Semaphores are used to block remote / padding writes until after
scratch and local writes are complete on all cores
- Existing kernel copy function was updated to perform stick by stick
copy when input / output chunks overlap and to perform copy in reverse
order when output data is moved “in front” of input data
- Local config was updated to reverse chunk order for “forward” copies
where dst is greated than src
- Coverage for this feature is included in MaxPool post-commit test on
WH and BH

### Checklist
- [x] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes:
https://github.com/tenstorrent/tt-metal/actions/runs/14129876766
- [x] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI passes (if applicable):
https://github.com/tenstorrent/tt-metal/actions/runs/14133031994,
https://github.com/tenstorrent/tt-metal/actions/runs/14136803225
- [x] [Model performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable):
https://github.com/tenstorrent/tt-metal/actions/runs/14132747748
- [x] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable):
https://github.com/tenstorrent/tt-metal/actions/runs/14129881697
- [x]
[Nightly](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable):
https://github.com/tenstorrent/tt-metal/actions/runs/14136743836
- [x] [N150 L2 nightly
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable):
https://github.com/tenstorrent/tt-metal/actions/runs/14178021568

---------

Co-authored-by: wransom-TT <wransom@tenstorrent.com>

afuller-TT committed:
[skip ci] 20.04 is dead. Long live 22.04 (#19931)

### Ticket
None

### Problem description
This job is failing b/c it's still trying to use 20.04, but with a 22.04
artifact.

### What's changed
Drop 20.04

blozano-tt committed:
[skip ci] Add temporary handling for 20.04 in build_metal.sh (#19940)

### Problem description
Some people want to keep working on 20.04 machines for a while.

### What's changed
If distro is detected as Ubuntu 20.04, use the old libc++ toolchain.

### Checklist
- [ ] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
- [ ] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI passes (if applicable)
- [ ] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [ ] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
CI passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

SeanNijjar committed:
Update fabric CODEOWNERS (#19937)

rfurko-tt committed:
[tt-train] Tensor parallel LLaMA (#19784)

### Problem description
Ability to run tensor paralleled llama.

### What's changed
* Add DistributedLLaMABlock
* Add DistributedLLaMAMLP
* Add DistributedGroupedQueryAttention

### Comparison

![image](https://github.com/user-attachments/assets/8c55aea7-5956-43c9-a91c-ca016bbe36de)

### Checklist
- [x] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
https://github.com/tenstorrent/tt-metal/actions/runs/14121773097
- [x] New/Existing tests provide coverage for changes

pprajapatiTT committed:
Merge branch 'pprajapati-19490' of https://github.com/tenstorrent/tt-metal into pprajapati-19490

pprajapatiTT committed:
Add support of random weights and reading kv cache in HF

Signed-off-by: Pratikkumar Prajapati <pprajapati@tenstorrent.com>

pprajapatiTT committed:
Add support of huggingface models and adjust PCC for mistral

Signed-off-by: Pratikkumar Prajapati <pprajapati@tenstorrent.com>

pprajapatiTT commented:
Opened new PR https://github.com/tenstorrent/tt-metal/pull/19995, as this PR got issues with rebase.
